rest/src/test/scala/org/apache/gearpump/services/AppMasterServiceSpec.scala

```diff
@@ -44,6 +46,16 @@ class AppMasterServiceSpec extends FlatSpec with ScalatestRouteTest with AppMast
       null
   }
 
+  def dag(config: Option[UserConfig]): DAG = {
+    config match {
```

Comment:
It is not OK to define this operation here. 
As we set the DAG field of UserConfig in AppDescription, you should also define this under AppDescription. 

A better way is to define a implicit conversion from Application to AppDescription in companion of AppDescription.

like this:

object AppDescription　｛
　def ApplicationToAppDescripton: AppDescription(app: Application) {
     val dag = _config.getValue[Graph[TaskDescription, Partitioner]](AppDescription.DAG).get)
    AppDescription(name, mainClass, dag)
  }
｝